tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/amqp/connect/AMQPFederationQueuePolicyTest.java

```diff
@@ -2782,6 +2785,8 @@ public void testPullQueueConsumerBatchCreditTopUpAfterEachBacklogDrain() throws
                                     .now();
             }
 
+            Wait.assertTrue(() -> server.queueQuery(queueName).getMessageCount() == DEFAULT_PULL_CREDIT_BATCH_SIZE);
```

Comment:
shouldn't it be >=?



Also, I have been using timeouts on all the assertEquals and assertTrues I have been adding.


I have been procrastinating removing the default method and always requiring a timeout set. as an eventual failure here would take 30seconds to happen.

Reply:
I suppose it could be although the test only sends DEFAULT_PULL_CREDIT_BATCH_SIZE message so being larger than that would indicate an intrinsic failure on the part of the broker to queue the messages sent in a rational manner.

Reply:
Setting a different timeout assumes one can guess how slow / fast the CI machine is which seems tough given the failures I've been seeing.  Personally I tend to be more concerned with the lengthy amount of time it takes to run the existing tests that do pass rather than the occasional failures due to timeout.  

I'd recommend lowering the default timeout value if that is of a major concern, drop it to 5 or 10 seconds as a baseline.

Reply:
@tabish121 I tried that and it wasn't simple for some reason I don't remember now.. I think some tests were failing. 

I talked to Robbie about that and we thought about removing the default timeout method.


this is a different discussion of course.